### DISCUSSION TITLE
Opting in to the new decorator Argument Marshalling syntax

### DISCUSSION THREAD

**Discussion by MaggieKimani1** [microsoft/typespec#3628] [General]
Hi, 
I'm currently working on a Typespec for copilot project that uses the OpenApi emitter to transform a `.tsp` file with custom decorators to the corresponding OpenApi description.
I have defined a custom decorator called state that takes in an object as one of the params;

#### State decorator syntax:
```typescript
 @state(function, state, { description , instructions, examples })
```
With the new upgrade, this syntax should ideally change to this:
```typescript
 @state(function, state, #{ description , instructions, examples })
```
I've tried upgrading the typespec libraries to the latest version  `0.57` and opting in to the new decorator marshalling behavior by setting the `valueMarshalling` package flag to `new` in my .ts file as per the [release notes](https://typespec.io/docs/next/release-notes/release-2024-06-10#typespeccompiler) but it doesn't seem to work.

#### Screenshots
![image](https://github.com/microsoft/typespec/assets/36787645/9bebcdb1-c05f-4241-b0f3-ed9b3245bbb8)

![image](https://github.com/microsoft/typespec/assets/36787645/e351289b-4c91-4cc2-85f9-20cec09de663)

Is there an extra step I'm missing?

**Comment by bterlson** [comment]
It seems like your language service is using an older version of TypeSpec without the value syntax. Can you ensure that you have 0.57 installed in your project, and don't have multiple TypeSpec versions floating around? You can also try updating your extension to 057 as well.

**Reply by bterlson** [reply]
Also fwiw wouldn't expect an augment decorator (`@@`) to work in that position, I think it has to be a top-level statement. You would probably get that error next after your language server understands the new value syntax.
